src/api/helpers/index.ts

```diff
@@ -1,29 +1,37 @@
-import { DEVELOPMENT, EXTENSION_ID } from "statics";
+import {
+  DEVELOPMENT,
+  EXTERNAL_MSG_REQUEST,
+  EXTERNAL_MSG_RESPONSE,
+} from "statics";
+import { Response } from "../types";
 
-interface Response {
-  applicationState: string;
-  publicKey: string;
-  mnemonicPhrase: string;
-  isCorrectPhrase: boolean;
-  confirmedPassword: boolean;
-  transactionStatus: string;
-  error: string;
-}
-
-export const sendMessageAndAwaitResponseExternal = (msg: {}): Promise<Response> =>
-  new Promise((resolve) => {
-    chrome.runtime.sendMessage(EXTENSION_ID, msg, (res: Response) =>
-      resolve(res),
-    );
+export const sendMessageToContentScript = (msg: {}): Promise<Response> => {
+  window.postMessage(
+    { source: EXTERNAL_MSG_REQUEST, ...msg },
+    window.location.origin,
+  );
```

Comment:
the idea is for the client site to post a message to itself. content script is listening and takes the message and sends it on to Background

Reply:
https://developer.chrome.com/extensions/messaging